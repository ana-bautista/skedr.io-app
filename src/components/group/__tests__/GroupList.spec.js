import { shallowMount } from '@vue/test-utils'
import component from '../GroupTagDialog'
import Vuetify from 'vuetify'
import Vue from 'vue'

Vue.use(Vuetify)
const createComp = propsData => shallowMount(component, { propsData })
describe('GroupTagDialog component', () => {
  describe('Computed properties', () => {
    it('has no new tag', () => {
      const wrapper = createComp({ title: '', group: { groupId: '123@N08' }, tags: ['firstTag'], manageTags: true })
      expect(wrapper.vm.canSave).toBe(false)
    })
    it('adds a tag', () => {
      const wrapper = createComp({ title: '', group: { groupId: '123@N08' }, tags: ['firstTag'], manageTags: true })
      wrapper.vm.comboTags.push('something')
      expect(wrapper.vm.canSave).toBe(true)
    })
    it('adds a new tag and deletes the previous one', () => {
      const wrapper = createComp({ title: '', group: { groupId: '123@N08' }, tags: ['firstTag'], manageTags: true })
      wrapper.vm.comboTags = ['secondTag']
      expect(wrapper.vm.canSave).toBe(true)
    })
    it('deletes all tags from list', () => {
      const wrapper = createComp({ title: '', group: { groupId: '123@N08' }, tags: ['firstTag'], manageTags: true })
      wrapper.vm.comboTags = []
      expect(wrapper.vm.canSave).toBe(true)
    })
  })
})
