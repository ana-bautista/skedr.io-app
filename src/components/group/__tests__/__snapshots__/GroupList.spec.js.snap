// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupList.vue shows a group 1`] = `
<div>
  <!---->
  <v-list-tile-stub activeclass="primary--text" avatar="true">
    <v-list-tile-avatar-stub size="40">
      <external-link-badge-stub groupid="groupId" icon=""></external-link-badge-stub>
    </v-list-tile-avatar-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>
        <photo-limit-opt-out-message-stub optout="true"></photo-limit-opt-out-message-stub> <strong>Group Title</strong>
      </v-list-tile-title-stub>
      <v-list-tile-sub-title-stub>
        <v-layout-stub tag="div">
          <v-flex-stub tag="div" xs2="" class="text-xs-left">
            <v-icon-stub>photo</v-icon-stub>
            1000M
          </v-flex-stub>
          <v-flex-stub tag="div" xs2="" class="text-xs-left">
            <v-icon-stub>group</v-icon-stub>
            200
          </v-flex-stub>
          <v-flex-stub tag="div" xs2="" class="text-xs-left pt-1"></v-flex-stub>
          <v-flex-stub tag="div" class="text-xs-right">
            <v-chip-stub value="true" label="true" selected="true" small="true">
              <v-icon-stub color="accent" small="" left="" class="mr-1">label</v-icon-stub>
              tag1
            </v-chip-stub>
            <v-chip-stub value="true" label="true" small="true">
              <v-icon-stub small="" left="" class="mr-1">label</v-icon-stub>
              tag2
            </v-chip-stub>
            <v-chip-stub value="true" label="true" small="true">
              <v-icon-stub small="" left="" class="mr-1">label</v-icon-stub>
              tag3
            </v-chip-stub>
          </v-flex-stub>
        </v-layout-stub>
      </v-list-tile-sub-title-stub>
    </v-list-tile-content-stub>
    <v-list-tile-action-stub>
      <v-btn-stub color="grey lighten-3" activeclass="v-btn--active" ripple="true" tag="button" icon="true" type="button">
        <v-tooltip-stub opendelay="200" closedelay="200" lazy="true" contentclass="" fixed="true" top="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" debounce="0" tag="span">
          <v-icon-stub color="primary">settings</v-icon-stub> <span>Manage tags</span>
        </v-tooltip-stub>
      </v-btn-stub>
    </v-list-tile-action-stub>
    <group-tag-dialog-stub groupid="groupId" title="Group Title" tags="tag1,tag2,tag3"></group-tag-dialog-stub>
  </v-list-tile-stub>
</div>
`;
